ImportExport/ObjectsToPython/ObjectsToPython.FCMacro

```diff
@@ -0,0 +1,344 @@
+            
+
+def addObject(doc, obj, objectlist): 
+    objname = varname(obj)
+    addScript("%s = doc.addObject('%s', '%s')"%(objname, obj.TypeId, objname))
+    defaultobj = doc.addObject(obj.TypeId, obj.Name + "Default")
+    addProperties(obj, objname, defaultobj, objectlist)
+    addProperties(obj.ViewObject, objname + ".ViewObject", defaultobj.ViewObject, [])                      
+    doc.removeObject(obj.Name + "Default")      
+    
+    
+def addProperties(obj, objname, refobj, objectlist):
+    skipProperties = ["Label", "Shape", "Proxy", "AddSubShape"]
+    skipObjProp = [("Sketcher::SketchObject", "Geometry"), ("Sketcher::SketchObject", "Constraints"),
+        ('PartDesign::Body', 'Origin'), ('PartDesign::Body', 'Tip')]
+    
+    if obj.TypeId == "Spreadsheet::Sheet":
+        addSpreadsheet(obj, objname)
+        return
+    
+    if obj.TypeId == "Sketcher::SketchObject":
+        addSketch(obj, objname)
+
+    for propname in obj.PropertiesList:
+        if not propname in skipProperties and not (obj.TypeId, propname) in skipObjProp:
+            prop = obj.getPropertyByName(propname)  
+            val = objectToText(prop, objectlist)
+            
+            try:
+                refprop = refobj.getPropertyByName(propname)
+                defaultval = objectToText(refprop, objectlist)
+            except:
+                defaultval = None       
+                
+            if propname == "Support":
+                val = val.replace("XY_Plane", "doc.XY_Plane")
+                val = val.replace("XZ_Plane", "doc.XZ_Plane")
+                val = val.replace("YZ_Plane", "doc.YZ_Plane")
+                
+            if propname == "ExpressionEngine":
+                for expression in prop:
+                    addScript("%s.setExpression%s"%(objname, expression))  
+                    
+            elif propname == "_Body":
+                addScript("%s_Group.append(%s)"%(val, objname))
+                          
+            elif (obj.TypeId, propname) == ('PartDesign::Body', 'Group'):
+                addScript("%s_Group = []"%(objname))
+                                                                                     
+            elif objectToText(prop, objectlist) is not None:   
+                if val != defaultval:        
+                    addScript("%s.%s = %s"%(objname, propname, val))   
+                                      
+                                                                                                                            
+def objectToText(obj, objectlist = []):    
+    if obj in objectlist:
+        return obj.Label
+    
+    else:
+        if hasattr(obj, 'Value') and hasattr(obj, 'Unit'):  # how can I test if type(obj) is Base.Quantity ?
+            return "%s"%(obj.Value)
+        else:     
+            if isinstance(obj, str):
+                return "\"%s\""%(obj)
+                
+            if isinstance(obj, bool):
+                if obj:
```

Comment:
`return str(obj)`

Reply:
This is marked as resolved but without change nor comment...

Reply:
sorry. fixed it.